crates/oxide/src/extractor/css_variable_machine.rs

```diff
@@ -0,0 +1,230 @@
+                    //            ^^
+                    Class::AllowedCharacter | Class::Dash | Class::Escape => cursor.advance(),
+
+                    // Valid character followed by anything else means the variable is done
+                    //
+                    // E.g.: `'--my-variable'`
+                    //                      ^
+                    _ => {
+                        // There must be at least 1 character after the `--`
+                        if cursor.pos - start_pos < 2 {
+                            return self.restart();
+                        } else {
+                            return self.done(start_pos, cursor);
+                        }
+                    }
+                },
+
+                Class::Escape => match CLASS_TABLE[cursor.next as usize] {
+                    // An escaped whitespace character is not allowed
+                    //
+                    // In CSS it is allowed, but in the context of a class it's not because then we
+                    // would have spaces in the class.
+                    //
+                    // E.g.: `bg-(--my-\ color)`
+                    //                  ^
+                    Class::Whitespace => return self.restart(),
+
+                    // An escape at the end of the class is not allowed
+                    Class::End => return self.restart(),
+
+                    // An escaped character, skip the next character, resume after
+                    //
+                    // E.g.: `--my-\#variable`
+                    //             ^           We are here
+                    //               ^         Resume here
+                    _ => cursor.advance_by(2),
+                },
+
+                // Character is not valid anymore
+                _ => return self.restart(),
+            }
+        }
+
+        MachineState::Idle
+    }
+}
+
+#[derive(Clone, Copy, PartialEq)]
+enum Class {
+    /// -
+    Dash,
+
+    /// _, a-z, A-Z, 0-9
+    AllowedCharacter,
+
+    /// \
+    Escape,
+
+    /// Whitespace
+    Whitespace,
+
+    /// End of the input
+    End,
+
+    Other,
+}
+
+const CLASS_TABLE: [Class; 256] = {
+    let mut table = [Class::Other; 256];
+
+    macro_rules! set {
+        ($class:expr, $($byte:expr),+ $(,)?) => {
+            $(table[$byte as usize] = $class;)+
+        };
+    }
+
+    macro_rules! set_range {
+        ($class:expr, $start:literal ..= $end:literal) => {
+            let mut i = $start;
+            while i <= $end {
+                table[i as usize] = $class;
+                i += 1;
+            }
+        };
+    }
+
+    set!(Class::Dash, b'-');
+    set!(Class::Escape, b'\\');
+    set!(Class::Whitespace, b' ', b'\t', b'\n', b'\r', b'\x0C');
+
+    set!(Class::AllowedCharacter, b'_');
+    set_range!(Class::AllowedCharacter, b'a'..=b'z');
+    set_range!(Class::AllowedCharacter, b'A'..=b'Z');
+    set_range!(Class::AllowedCharacter, b'0'..=b'9');
+
```

Comment:
Might need to include extended unicode ranges here too: https://play.tailwindcss.com/8nUkRqhgvb

Reply:
Okay I just checked and this does work v4 right now: https://play.tailwindcss.com/H7MlnsHqpy 😅

Reply:
https://github.com/tailwindlabs/tailwindcss/commit/21fe01922248c6e5d8cef15de35559aae8d3ccd0 😎